app/src/features/sessionBook/screens/DraftSessionScreen/components/SessionTrimmer/SessionTrimmer.tsx

```diff
@@ -137,6 +92,52 @@ export const SessionTrimmer: React.FC<SessionTrimmerProps> = ({ duration, sessio
     dispatch(sessionRecordingActions.setTrimmedSessiondata(trimmedData));
   }, [trimSessionEvents, dispatch]);
 
+  const debouncedTrimSessionEvents = useDebounce(handleTrim, 200);
+
+  const handleDrag = useCallback(
+    (event: MouseEvent, isStart: boolean) => {
+      if (!containerRef.current) return;
+      const rect = containerRef.current.getBoundingClientRect();
+      const x = event.clientX - rect.left;
+      const newTime = Math.round((x / rect.width) * duration);
+      if (!isSessionTrimmed) {
+        setIsSessionTrimmed(true);
+      }
+      if (isStart) {
+        setStartTime(Math.max(0, Math.min(newTime, endTime - 1)));
+      } else {
+        setEndTime(Math.min(duration, Math.max(newTime, startTime + 1)));
+      }
+      debouncedTrimSessionEvents();
+    },
+    [duration, startTime, endTime, isSessionTrimmed, debouncedTrimSessionEvents]
+  );
+
+  const startDrag = useCallback(
+    (isStart: boolean) => {
+      const handleMouseMove = (e: MouseEvent) => handleDrag(e, isStart);
+      const handleMouseUp = () => {
+        document.removeEventListener("mousemove", handleMouseMove);
+        document.removeEventListener("mouseup", handleMouseUp);
+      };
+
+      document.addEventListener("mousemove", handleMouseMove);
+      document.addEventListener("mouseup", handleMouseUp);
```

Comment:
as discussed, `handleMouseUp` being passed forward inside `handleMouseUp` seems wrong. That removeListener should probably be shifted to the useEffect to fix player pause state bug (_possibly_)